server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -1307,9 +1316,27 @@ public Builder put(IndexMetadata indexMetadata, boolean incrementVersion) {
             if (unsetPreviousIndicesLookup(previous, indexMetadata)) {
                 previousIndicesLookup = null;
             }
+            maybeSetMappingPurgeFlag(previous, indexMetadata);
             return this;
         }
 
+        private void maybeSetMappingPurgeFlag(@Nullable IndexMetadata previous, IndexMetadata updated) {
+            if (previous == null) {
```

Comment:
Can we shortcut here if `checkForUnusedMappings` is `true`?

Reply:
++ sure, 85183dd :)